wpiutil/src/main/native/cpp/uv/ParallelRepeatTCPConnector.cpp

```diff
@@ -0,0 +1,126 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "wpi/uv/ParallelRepeatTCPConnector.h"
+
+namespace wpi {
+namespace uv {
+
+void ParallelRepeatTCPConnector::Start() {
+  auto async = Async<>::Create(m_loop);
+  m_asyncEnter = async;
+  async->wakeup.connect([self = shared_from_this()] {
```

Comment:
It seems like this shouldn't have async stuff in it.  async is only needed when communicating between threads, and this code isn't threaded.  If there's stuff outside the event loop it's that code's responsibility to handle the async call into the event loop.

Reply:
The async was the only way I could find to set the IP addresses to connect to. I atttempted to do it outside and it ended up way more ugly. I also use the async as a handle to know when the loop is shutting down, as otherwise there is no other way to tell.